chalice/local.py

```diff
@@ -109,98 +128,254 @@ def create_lambda_event(self, method, path, headers, body=None):
+        if not route_entry:
+            return lambda_event, lambda_context
+        authorizer = route_entry.authorizer
+        if not authorizer:
+            return lambda_event, lambda_context
+        if not isinstance(authorizer, ChaliceAuthorizer):
+            # Currently the only supported local authorizer is the
+            # BuiltinAuthConfig type. Anything else we will err on the side of
+            # allowing local testing by simply admiting the request. Otherwise
+            # there is no way for users to test their code in local mode
+            warnings.warn((
+                '%s is not a supported in local mode. All requests made '
+                'against a route will be authorized to allow local testing.')
+                % authorizer.__class__.__name__
+            )
+            return lambda_event, lambda_context
+        auth_result = authorizer(auth_event, lambda_context)
+        authed = self._check_can_invoke_view_function(
+            route_entry, auth_result)
+        if authed:
+            lambda_event = self._update_lambda_event(lambda_event, auth_result)
+        else:
+            raise NotAuthorizedError(
+                {'x-amzn-RequestId': lambda_context.aws_request_id,
+                 'x-amzn-ErrorType': 'UnauthorizedException'})
+        return lambda_event, lambda_context
+
+    def _check_can_invoke_view_function(self, route_entry, auth_result):
+        # type: (RouteEntry, ResponseType) -> bool
+        policy = auth_result.get('policyDocument', {})
+        statements = policy.get('Statement', [])
+        # Look for a statement that allows us to invoke the route_entry
+        # it will be in the form local:local/METHOD/uri the URI is always
+        # prefixed with / so we do not need to explicitly include it in our
+        # pattern.
+        # The method can be either an explicit method such as POST, GET, etc...
+        # or it can be a *. We also need to match for any prefix since the
+        # user can return any policy at all from the custom authorizer.
+        pattern_suffix = ''
+        if route_entry.uri_pattern == '/':
+            # There is a special case when the / is used since the API Gateway
+            # arn will have the / duplicated. On all other routes it will
+            # omit the leading /.
+            pattern_suffix = '/'
+        method_pattern = r'(?:\*|%s)' % route_entry.method
+        route_entry_pattern = r'^.*?/%s%s%s$' % (method_pattern,
+                                                 route_entry.uri_pattern,
+                                                 pattern_suffix)
+        route_entry_re = re.compile(route_entry_pattern)
+        # Look through all statements for one that allows us to invoke this
+        # particular view function.
+        for statement in statements:
+            if statement.get('Effect') == 'Allow' and \
+               statement.get('Action') == 'execute-api:Invoke':
```

Comment:
These technically support wildcards as well.  If you're not going to support them, you should probably make a note about eventually supporting them.